cancel-in-progress/parent_workflow.go

```diff
@@ -0,0 +1,109 @@
+package cancel_in_progress
+
+import (
+	"errors"
+	"go.temporal.io/sdk/log"
+	"go.temporal.io/sdk/workflow"
+)
+
+const ParentWorkflowSignalName = "parent-workflow-signal"
+
+func SampleParentWorkflow(ctx workflow.Context) (result string, err error) {
+	logger := workflow.GetLogger(ctx)
+
+	var message string
+
+	reBuildSignalChan := workflow.GetSignalChannel(ctx, ParentWorkflowSignalName)
+
+	// This will not block because the workflow is started with a signal transactional.
+	reBuildSignalChan.Receive(ctx, &message)
+
+	cwo := workflow.ChildWorkflowOptions{
+		WaitForCancellation: true,
+	}
+	ctx = workflow.WithChildOptions(ctx, cwo)
+
+	var isProcessingDone = false
+
+	for !isProcessingDone {
+		ctx, cancelHandler := workflow.WithCancel(ctx)
+
+		// it is important to re-execute the child workflow after every signal
```

Comment:
But you don't execute after every signal. If you get two signals you only execute after the latest. (seems to be by intention, just need to change this comment)